### PR TITLE
Issue #1604: Tweak handling of keys that failed verification in `mod_…

### PR DESCRIPTION
**Pull request by Castaglia**
…sftp_sql`, such that we fall back to attempting verification of the keys as "raw" encoding only if the verification failure was indeed do to format/parsing issues, _vs_ key mismatches.

### CODE REVIEW

File: contrib/mod_sftp_sql.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ struct sqlstore_data {
 
 static const char *sqlstore_user = NULL;
 
-static const char *trace_channel = "ssh2";
+static const char *trace_channel = "sftp.sql";
```
Comment (Castaglia) [inline contrib/mod_sftp_sql.c:53]: Note that I created a separate trace channel for this module.  I've found that, as I develop more modules, having focused trace channels whose names align more closely with their module makes the logging easier to follow.

File: contrib/mod_sftp_sql.c [hunk 2]
```diff
@@ -465,23 +465,29 @@ static int sqlstore_verify_key_raw(pool *p, struct sqlstore_data *store_data,
       store_data->select_query, strerror(errno));
 
   } else if (res == FALSE) {
-    (void) pr_log_writefile(sftp_logfd, MOD_SFTP_SQL_VERSION,
+    pr_trace_msg(trace_channel, 3,
```
Comment (Castaglia) [inline contrib/mod_sftp_sql.c:468]: These "key does not match" log messages might be a bit noisy for `SFTPLog`, so I've moved them to trace logging.

**Comment by offsides** [comment]
This looks good.  The one thing I will point out is that mod_sftp is still suggesting that the key is not RFC4716 formatted when it doesn't match, in mod_sftp/keystore.c on line 246.  I'm seeing the following in my logs:
```
2023-02-24 11:28:03,613 mod_sftp/1.1.1[708547]: public key SHA256 fingerprint: d6:0f:e8:3f:c7:03:55:e1:8e:83:1d:07:e6:f7:7e:51:e8:ab:9f:e6:52:16:18:7e:08:b5:43:4b:91:93:41:58
2023-02-24 11:28:03,615 mod_sftp_sql/0.4[708547]: SQLNamedQuery 'get-user-authorized-keys' returned 1 row
2023-02-24 11:28:03,615 mod_sftp/1.1.1[708547]: no matching public key found for user 'sftptest2' in '/get-user-authorized-keys'; perhaps keys are not RFC4716-formatted
```
So mod_sftp_sql is properly decoding the key and reporting normally, but mod_sftp either can't tell that the key was formatted properly or isn't checking for that difference.  Not a big deal, but perhaps something to add to the rainy-day todo list. :)

Otherwise, I think this resolves the erroneous error about decoding raw keys nicely.  Thanks!

**Comment by Castaglia** [comment]
> This looks good. The one thing I will point out is that mod_sftp is still suggesting that the key is not RFC4716 formatted when it doesn't match, in mod_sftp/keystore.c on line 246. I'm seeing the following in my logs:
> ```
> 2023-02-24 11:28:03,613 mod_sftp/1.1.1[708547]: public key SHA256 fingerprint: d6:0f:e8:3f:c7:03:55:e1:8e:83:1d:07:e6:f7:7e:51:e8:ab:9f:e6:52:16:18:7e:08:b5:43:4b:91:93:41:58
> 2023-02-24 11:28:03,615 mod_sftp_sql/0.4[708547]: SQLNamedQuery 'get-user-authorized-keys' returned 1 row
> 2023-02-24 11:28:03,615 mod_sftp/1.1.1[708547]: no matching public key found for user 'sftptest2' in '/get-user-authorized-keys'; perhaps keys are not RFC4716-formatted
> ```
> 
> So mod_sftp_sql is properly decoding the key and reporting normally, but mod_sftp either can't tell that the key was formatted properly or isn't checking for that difference. Not a big deal, but perhaps something to add to the rainy-day todo list. :)

This is due to the separation of concerns.  Specifically, `mod_sftp` only knows that no matching key was found; it's the provider module (`mod_sftp_sql` in this case, but it could be some other keystore provider) that knows more about _why_ the matching/verification failed.

**Pull request merged**